connect-inject/handler.go

```diff
@@ -399,85 +315,53 @@ func (h *Handler) Mutate(req *v1beta1.AdmissionRequest) *v1beta1.AdmissionRespon
 	//	}
 	//}
 
-	patches = append(patches, addContainer(
-		pod.Spec.Containers,
-		[]corev1.Container{esContainer},
-		"/spec/containers")...)
-
-	// Add annotations so that we know we're injected
-	patches = append(patches, updateAnnotation(
-		pod.Annotations,
-		map[string]string{
-			annotationStatus: injected,
-		})...)
+	pod.Annotations[annotationStatus] = injected
 
 	// Add annotations for metrics
 	promAnnotations, err := h.prometheusAnnotations(&pod)
 	if err != nil {
 		h.Log.Error("Error configuring prometheus annotations", "err", err, "Request Name", req.Name)
-		return &v1beta1.AdmissionResponse{
-			Result: &metav1.Status{
-				Message: fmt.Sprintf("Error configuring prometheus annotations: %s", err),
-			},
-		}
+		return admission.Errored(http.StatusInternalServerError, fmt.Errorf("error configuring prometheus annotations: %s", err))
 	}
-	if promAnnotations != nil {
-		patches = append(patches, updateAnnotation(
-			pod.Annotations,
-			promAnnotations)...)
+	for key, value := range promAnnotations {
+		pod.Annotations[key] = value
 	}
 
-	// Add Pod label for health checks
-	patches = append(patches, updateLabels(
-		pod.Labels,
-		map[string]string{
-			labelInject: injected,
-		})...)
+	if pod.Labels == nil {
+		pod.Labels = make(map[string]string)
+	}
+	pod.Labels[labelInject] = injected
 
 	// Consul-ENT only: Add the Consul destination namespace as an annotation to the pod.
 	if h.EnableNamespaces {
-		patches = append(patches, updateAnnotation(
-			pod.Annotations,
-			map[string]string{
-				annotationConsulNamespace: h.consulNamespace(req.Namespace),
-			})...)
-	}
-
-	// Generate the patch
-	var patch []byte
-	if len(patches) > 0 {
-		var err error
-		patch, err = json.Marshal(patches)
-		if err != nil {
-			h.Log.Error("Could not marshal patches", "err", err, "Request Name", req.Name)
-			return &v1beta1.AdmissionResponse{
-				Result: &metav1.Status{
-					Message: fmt.Sprintf("Could not marshal patches: %s", err),
-				},
-			}
-		}
+		pod.Annotations[annotationConsulNamespace] = h.consulNamespace(req.Namespace)
+	}
+
+	updatedPodJson, err := json.Marshal(pod)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
 
-		resp.Patch = patch
-		patchType := v1beta1.PatchTypeJSONPatch
-		resp.PatchType = &patchType
+	patches, err := jsonpatch.CreatePatch(origPodJson, updatedPodJson)
```

Comment:
Nice, I like this over compiling the list of patches with each step!

Reply:
yeah I agree! looks much cleaner!